mautic/mautic
# WIP : Logging warnings while importing contacts or custom items

MAUT-11097 : Logging warnings while importing contacts or customitems



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

While importing a customItems with LinkedContactIds, If provided contactId is not valid or deleted from instance it is failing whole import process, hence the remaining records in csv are able to import.
Now, we are checking if provided contactId is valid or not and if it is not valid just log the warning and continue import process for remaining records in csv.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a csv for import CustomItems with LinkedContactIds
3. add some invalid contactIds for LinkedContactIds column in csv
4. Import the CustomItems for customObject.
5. Observe that all customItems should import and you can see the warnings for invalid contacts on import details page.
